### PR TITLE
add spinner to posts and replies while loading

### PR DESCRIPTION
**Pull request by claireolmstead**
Purpose
---------------
Feature
[https://www.pivotaltracker.com/story/show/179073406](url)

Solution
---------------
As a user, I want to see that my post or reply is loading after I post it.  create gray loading placeholder post/reply.

Change summary
---------------
* keep track of new post loading, new reply loading, and reply parent in the redux store
* create Blank Post and Blank Reply
* display the two above components based on the above redux state

Steps to Verify
----------------
1. create a post
2. create a reply
3. have multiple posts and multiple replies

Additional details / screenshot
----------------
*** Does this work if multiple people are posting are once?

https://user-images.githubusercontent.com/43625033/129978702-da5db0b6-becf-41b6-9e8c-069021f319ac.mov

### CODE REVIEW

File: src/components/Feed.tsx [hunk 1]
```diff
@@ -55,7 +55,9 @@ const Feed = (): JSX.Element => {
         )}
         {isModalOpen && (
           <NewPost
-            onSuccess={() => setIsModalOpen(false)}
```
Comment (claireolmstead) [inline src/components/Feed.tsx:58]: can change back

**Review by enddynayn — APPROVED** [review]

File: src/components/ReplyBlock.tsx [hunk 2]
```diff
@@ -15,13 +16,20 @@ const ReplyBlock = ({ parent }: ReplyBlockProps): JSX.Element => {
     return reply?.content?.type === "Note" && reply?.inReplyTo === parent;
   }) as FeedItem[];
 
+  const loading: any = useAppSelector((state) => state.feed.isReplyLoading);
```
Comment (claireolmstead) [inline src/components/ReplyBlock.tsx:19]: add type

File: src/redux/slices/feedSlice.ts [hunk 3]
```diff
@@ -33,7 +39,30 @@ export const feedSlice = createSlice({
...
       };
     },
+    postLoading: (state, isLoading: PayloadAction<boolean>) => {
+      return {
+        ...state,
+        feed: [...state.feed],
+        isPostLoading: isLoading.payload,
+      };
+    },
+    replyLoading: (state, isLoading: PayloadAction<any>) => {
```
Comment (claireolmstead) [inline src/redux/slices/feedSlice.ts:49]: change type

File: src/components/Profile.tsx [hunk 4]
```diff
@@ -46,12 +47,6 @@ const Profile = (): JSX.Element => {
 
   const saveEditProfile = async () => {
     setIsEditing(!isEditing);
```
Comment (shannonwells) [inline src/components/Profile.tsx:49]: Why is profile saving deleted?

File: src/components/NewPost.tsx [hunk 5]
```diff
@@ -47,6 +49,8 @@ const NewPost = ({ onSuccess, onCancel }: NewPostProps): JSX.Element => {
       userId
     );
     await sendPost(newPostFeedItem);
+    console.log(newPostFeedItem);
```
Comment (shannonwells) [inline src/components/NewPost.tsx:52]: remove console.log

File: src/redux/slices/feedSlice.ts [hunk 6]
```diff
@@ -15,6 +29,14 @@ export const feedSlice = createSlice({
   reducers: {
     addFeedItem: (state, action: PayloadAction<FeedItem>) => {
       const newFeedItem = action.payload;
+      if (state.isPostLoading.myIdentifier === newFeedItem.fromId) {
```
Comment (shannonwells) [inline src/redux/slices/feedSlice.ts:32]: Explanatory comment (as discussed)

File: src/redux/slices/feedSlice.ts [hunk 7]
```diff
@@ -1,12 +1,26 @@
...
 
 interface feedState {
   feed: FeedItem[];
+  isPostLoading: isPostLoadingType;
+  isReplyLoading: isReplyLoadingType;
 }
 
 const initialState: feedState = {
   feed: [],
+  isPostLoading: { loading: false, myIdentifier: undefined },
```
Comment (shannonwells) [inline src/redux/slices/feedSlice.ts:22]: Can we change myIdentifier to currentUserId?

**Review by shannonwells — CHANGES_REQUESTED** [review]
As discussed

**Review by shannonwells — APPROVED** [review]
💯 👍

**Pull request merged**